src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/RMSMappingQuality.java

```diff
@@ -31,32 +32,64 @@
+     */
+    @Override
+    public Map<String, Object> annotateRawData(final ReferenceContext ref,
+                                               final VariantContext vc,
+                                               final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap){
         Utils.nonNull(vc);
         if (perReadAlleleLikelihoodMap == null || perReadAlleleLikelihoodMap.isEmpty() ) {
-            return null;
+            return Collections.emptyMap();
         }
 
-        //get all available mapping qualities
-        final List<Integer> qualities = perReadAlleleLikelihoodMap.values().stream().
-                flatMap(pprl -> pprl.getReads().stream().map(read -> read.getMappingQuality()).filter(mq -> mq != QualityUtils.MAPPING_QUALITY_UNAVAILABLE))
-                .collect(Collectors.toList());
-        final double rms = MathUtils.rms(qualities);
-        return Collections.singletonMap(getKeyNames().get(0), formatedValue(rms));
+        final Map<String, Object> annotations = new HashMap<>();
+        final ReducibleAnnotationData<Number> myData = new ReducibleAnnotationData<>(null);
+        calculateRawData(vc, perReadAlleleLikelihoodMap, myData);
+        final String annotationString = formatedValue((double) myData.getAttributeMap().get(Allele.NO_CALL));
+        annotations.put(getRawKeyName(), annotationString);
+        return annotations;
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})//FIXME
+    public void calculateRawData(final VariantContext vc,
+                                 final Map<String, PerReadAlleleLikelihoodMap> stratifiedPerReadAlleleLikelihoodMap,
+                                 final ReducibleAnnotationData rawAnnotations){
+        if (stratifiedPerReadAlleleLikelihoodMap.isEmpty()) {
+            return;
+        }
+
+        //put this as a double, like GATK3.5
+        final double squareSum = stratifiedPerReadAlleleLikelihoodMap.values().stream()
+                .flatMap(likelihoodMap -> likelihoodMap.getReads().stream())
+                .map(GATKRead::getMappingQuality)
+                .filter(mq -> mq != QualityUtils.MAPPING_QUALITY_UNAVAILABLE).mapToDouble(mq -> mq * mq).sum();
+        rawAnnotations.putAttribute(Allele.NO_CALL, squareSum);
+    }
+
+    @Override
+    public Map<String, Object> annotate(final ReferenceContext ref,
+                                        final VariantContext vc,
+                                        final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap) {
+        return annotateRawData(ref, vc, perReadAlleleLikelihoodMap);
     }
 
     @VisibleForTesting
     static String formatedValue(double rms) {
         return String.format("%.2f", rms);
     }
 
-    public List<String> getKeyNames() { return Collections.singletonList(VCFConstants.RMS_MAPPING_QUALITY_KEY); }
+    @Override
+    public List<String> getKeyNames() { return Collections.singletonList(getRawKeyName()); }
```

Comment:
i found a workaround - return 2 keys here and 2 descriptions in getDescriptions. Works fine